perception/map_based_prediction/src/map_based_prediction_node.cpp

```diff
@@ -926,6 +989,19 @@ void MapBasedPredictionNode::objectsCallback(const TrackedObjects::ConstSharedPt
   // result debug
   visualization_msgs::msg::MarkerArray debug_markers;
 
+  // get current crosswalk users for later prediction
+  std::unordered_map<std::string, TrackedObject> current_crosswalk_users;
+  for (const auto & object : in_objects->objects) {
+    const auto label_for_prediction =
+      changeLabelForPrediction(object.classification.front().label, object, lanelet_map_ptr_);
+    if (
+      label_for_prediction == ObjectClassification::PEDESTRIAN ||
+      label_for_prediction == ObjectClassification::BICYCLE) {
+      current_crosswalk_users[tier4_autoware_utils::toHexString(object.object_id)] = object;
```

Comment:
I prefer creating a temp variable:
const auto object_id = tier4_autoware_utils::toHexString(object.object_id)
Also, please use the unordered_maps' insert method to insert the object into the map

Reply:
any reason to use insert and not `emplace` here? I can see that everywhere else in the code emplace is used

Reply:
emplace is also good @dkoldaev